clippy_lints/src/casts/unnecessary_cast.rs

```diff
@@ -167,11 +168,11 @@ pub(super) fn check<'tcx>(
                 sym::assert_ne_macro,
                 sym::debug_assert_ne_macro,
             ];
-            matches!(expr.span.ctxt().outer_expn_data().macro_def_id, Some(def_id) if 
+            matches!(expr.span.ctxt().outer_expn_data().macro_def_id, Some(def_id) if
                 cx.tcx.get_diagnostic_name(def_id).is_some_and(|sym| ALLOWED_MACROS.contains(&sym)))
         }
 
-        if let Some(id) = path_to_local(cast_expr)
+        if let Some(id) = cast_expr.res_local_id()
```

Comment:
Shouldn't it be `opt_res_local_id()` for consistency with `opt_diag_name()` and others?

Reply:
I originally had it as `opt_local_id`, but changed it to `res_local_id` to match all the other names to get a path's resolution. I'm not strongly against `opt_res_local_id`, but it's getting a little on the wordy side. I'm not really attached to many of the names here.

File you want to look at for the new functions is `clippy_utils/res.rs`. Just noticed this is 279 files. I knew it was a lot...